init.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+CONTAINER_READY="Startup_Init"
+if [ ! -e $CONTAINER_READY ]; then
+    if [ ! -e "/storage/.env"]; then
+        is_first_time()
+    fi
+else
+    not_first_time
+fi
+
+function is_first_time()
+{
+    touch $CONTAINER_READY
+    echo "Running first-time startup logic"
+    php artisan key:generate
+    php artisan migrate
```

Comment:
migrate and db:seed can be run every time the container starts, since it won’t re-run already executed commands. After each upgrade, this should be run, so it may be worth adding this to the startup for each launch, as well as a git pull. Else building an upgrade function in the web-UI is an option. Will consider

Reply:
I ended up ditching this logic, due to a bug. I'll try and revisit this to get this working so I can automate the initial/upgrade experience.

Reply:
Added all of these features (except the web ui stuff, that's not my department).